src/extension.ts

```diff
@@ -27,6 +27,7 @@ import { getAiKey, getExtensionId, getExtensionVersion, loadPackageInfo } from "
 import { executeInTerminal } from "./utils/mavenUtils";
 import { openFileIfExists, showTroubleshootingDialog } from "./utils/uiUtils";
 import { Utils } from "./utils/Utils";
+export let isJavaExtensionInstalled: boolean = false;
```

Comment:
use https://github.com/microsoft/vscode-maven/blob/b78d316f92b0df4cf73e75b0ed76c056869d00bc/src/jdtls/commands.ts#L19

add a local variable there to cache the value. e.g.

```ts
let isJavaExtensionInstalled: boolean | undefined;
export function isJavaExtEnabled() {
  if (isJavaExtensionInstalled !== undefined) return isJavaExtensionInstalled;
  ...
}
```